public/app/plugins/panel/heatmap/HeatmapHoverView.tsx

```diff
@@ -120,7 +125,14 @@ const HeatmapHoverCell = ({ data, hover, showHistogram }: Props) => {
   const linkLookup = new Set<string>();
 
   for (const field of visibleFields ?? []) {
-    // TODO: Currently always undefined? (getLinks)
+    const hasLinks = field.config.links && field.config.links.length > 0;
+    if (hasLinks && data.heatmap) {
+      let appropriateScopedVars = scopedVars.filter(
+        (sv) => sv && sv.__dataContext && sv.__dataContext.value.field.name === nonNumericOrdinalDisplay
+      )[0];
+      field.getLinks = getLinksSupplier(data.heatmap, field, appropriateScopedVars ?? {}, replaceVars);
+    }
```

Comment:
nit - but maybe something more along the lines of

```suggestion
const hasLinks = field.config.links && field.config.links.length > 0;

if (hasLinks && data.heatmap) {
  const appropriateScopedVars = scopedVars.find((scopedVar) => scopedVar && scopedVar.__dataContext && scopedVar.__dataContext.value.field.name === nonNumericOrdinalDisplay);

  field.getLinks = getLinksSupplier(data.heatmap, field, appropriateScopedVars || {}, replaceVars);
}
```